File: java_solutions/src/com/mybasepackage/MaxProfit.java
package com.mybasepackage;

class MaxProfit {
    public int maxProfit(int[] prices) {
        int maxProfit = 0;
        int i = 0;
        while (i<prices.length-1) {
            int current_price = prices[i];
            int future_price = prices[i+1];
            if (current_price < future_price) {
                maxProfit += (future_price - current_price);
            }
            i++;
        }
        return maxProfit;
    }

    public static void main(String[] args) {
        MaxProfit solution = new MaxProfit();
        int[] prices = new int[]{1,2,3,4,5};
        int maxProfit = solution.maxProfit(prices);
        System.out.println(String.format("Max profit: %s", maxProfit));
    }
}

File: python_solutions/easy/arrays/max_profit.py
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """

        i = 0
        max_profit = 0
        while i < len(prices) - 1:
            current_price = prices[i]
            future_price = prices[i + 1]

            if current_price < future_price:
                max_profit += (future_price - current_price)
            i += 1

        return max_profit


prices = [1, 2, 3, 4, 5]
max_profit = Solution().maxProfit(prices)
print("max profit: %s" % max_profit)
